config/app/api/openapi/registry.yml

```diff
@@ -0,0 +1,127 @@
+                default:
+                    $ref: '#/components/responses/ErrorResponseDefault'
+        put:
+            summary: 'Updates application configuration between Tenant and App'
+            parameters:
+                -   $ref: '#/components/parameters/storeReference'
+            requestBody:
+                $ref: '#/components/requestBodies/ApplicationConfigurationRequestBody'
+            responses:
+                200:
+                    $ref: '#/components/responses/ApplicationConfigurationResponseStatus200'
+                400:
+                    $ref: '#/components/responses/BadRequestResponseDefault'
+                default:
+                    $ref: '#/components/responses/ErrorResponseDefault'
+
+    '/private/application/disconnect':
+        post:
+            summary: 'Disconnects Application from a Tenants store.
+                      Finds Application configuration and removes it.'
+            parameters:
+                -   $ref: '#/components/parameters/storeReference'
+            responses:
+                204:
+                    description: 'App is disconnected.'
+                400:
+                    $ref: '#/components/responses/BadRequestResponseDefault'
+                default:
+                    $ref: '#/components/responses/ErrorResponseDefault'
+components:
+    schemas:
+        ConfigurationApiResponse:
+            type: object
+            additionalProperties: true
+            example:
+                configKey1: "config value"
+                configKey2: ["selectedValue1", "selectedValue2"]
+        ConfigurationApiRequest:
+            properties:
+                data:
+                    type: object
+                    required:
+                        - attributes
+                    properties:
+                        attributes:
+                            type: object
+                            required:
+                                - configuration
+                            properties:
+                                configuration:
+                                    description: 'Contains JSON encoded configuration.'
+                                    type: string
+                                    example: "{\"configKey1\":\"config value\",\"configKey2\":[\"selectedValue1\",\"selectedValue2\"]}"
+        ApiErrorMessage:
+            properties:
+                code:
+                    type: string
+                detail:
+                    type: string
+                status:
+                    type: string
+        ApiErrorMessages:
+            properties:
+                data:
+                    type: array
+                    items:
+                        $ref: '#/components/schemas/ApiErrorMessage'
+    requestBodies:
+        ApplicationConfigurationRequestBody:
+            description: 'Provides configuration for the application.'
+            required: true
+            content:
+                application/json:
```

Comment:
Please check all content types. When we have

```
{
  "data": {
    "attributes": {
      "configuration": "{\"configKey1\":\"config value\",\"configKey2\":[\"selectedValue1\",\"selectedValue2\"]}"
    }
  }
}
```
It must be `application/vnd.api+json`.

@k4emic @matweew I see a lot mixed content types. Shouldn't we stick for all to `application/vnd.api+json`? At least for our "normal" requests/responses. For errors I'm not sure TBH.

Reply:
Per my understanding, `application/vnd.api+json` is only used if you're following the [jsonapi.org](https://jsonapi.org/) standard (a JSON API design convention of sorts).

> JSON:API has been properly registered with the IANA. Its media type designation is [application/vnd.api+json](http://www.iana.org/assignments/media-types/application/vnd.api+json).

For normal JSON APIs (like ours) we should stick with `application/json` (the petstore example, and the openapi spec examples uses this, too).

Reply:
AFAIK we should use the JSON API. And it's reflected in the structure of `data: attributes:`

Reply:
We're using something that resembles the JSON:API spec for the body, but we're not running a real JSON:API-compatible specification.

For example, we do not adhere to these requirements (and probably more, I only spent a few minutes picking through the spec)

> Every [resource object](https://jsonapi.org/format/#document-resource-objects) MUST contain an id member and a type member. The values of the id and type members MUST be strings.

> Servers MUST send all JSON:API data in response documents with the header Content-Type: application/vnd.api+json without any media type parameters.

I'm 100% for using JSON:API as our preferred schema, but doing so is beyond the scope of what we're trying to do here.

Reply:
In this case I won't be doing any changes to content-type